### DISCUSSION TITLE
Different results after babel build

### DISCUSSION THREAD

**Discussion by philkau** [babel/babel#11720] [Q&A]
I have two Windows environments. I get the different ways to export default module.

First: **exports.default = void 0;**

Second: **exports["default"] = void 0;**

I am using the Babel v7.8.4.

The build script is **babel ./src --out-dir ./dist** 

The .babelrc is:
```json
{
  "presets": [
    "@babel/preset-env",
    "@babel/preset-react"
  ],
  "plugins": [
    "@babel/plugin-proposal-object-rest-spread"
  ]
}
```
Any ideas?

**Comment by JLHwung** [comment]
Can you attach the output of
```diff
{
  "presets": [
---    "@babel/preset-env",
+++    ["@babel/preset-env", { "debug": true }],
    "@babel/preset-react"
  ],
  "plugins": [
    "@babel/plugin-proposal-object-rest-spread"
  ]
}
```
It will print the active plugin applied by `preset-env`.

**Reply by philkau** [reply]
Hi @JLHwung,

Thanks~

The two environments use the different plugins.
BUT I could not find where I could set the variable.
Could you give me a favor?

These are the results.

### FIRST:
```
Using plugins:
  proposal-nullish-coalescing-operator {}
  proposal-optional-chaining {}
  proposal-json-strings {}
  proposal-optional-catch-binding {}
  transform-parameters {}
  proposal-async-generator-functions {}
  proposal-object-rest-spread {}
  transform-dotall-regex {}
  proposal-unicode-property-regex {}
  transform-named-capturing-groups-regex {}
  transform-async-to-generator {}
  transform-exponentiation-operator {}
  transform-template-literals {}
  transform-literals {}
  transform-function-name {}
  transform-arrow-functions {}
  transform-block-scoped-functions {}
  transform-classes {}
  transform-object-super {}
  transform-shorthand-properties {}
  transform-duplicate-keys {}
  transform-computed-properties {}
  transform-for-of {}
  transform-sticky-regex {}
  transform-unicode-escapes {}
  transform-unicode-regex {}
  transform-spread {}
  transform-destructuring {}
  transform-block-scoping {}
  transform-typeof-symbol {}
  transform-new-target {}
  transform-regenerator {}
  transform-member-expression-literals {}
  transform-property-literals {}
  transform-reserved-words {}
  transform-modules-commonjs {}
  proposal-dynamic-import {}
```

### SECOND:
```
Using plugins:
  proposal-nullish-coalescing-operator { "chrome":"49", "edge":"18", "firefox":"68", "ie":"11", "ios":"12", "safari":"12.1", "samsung":"10.1" }
  proposal-optional-chaining { "chrome":"49", "edge":"18", "firefox":"68", "ie":"11", "ios":"12", "safari":"12.1", "samsung":"10.1" }
  proposal-json-strings { "chrome":"49", "edge":"18", "ie":"11" }
  proposal-optional-catch-binding { "chrome":"49", "edge":"18", "ie":"11" }
  transform-parameters { "ie":"11" }
  proposal-async-generator-functions { "chrome":"49", "edge":"18", "ie":"11" }
  proposal-object-rest-spread { "chrome":"49", "edge":"18", "ie":"11" }
  transform-dotall-regex { "chrome":"49", "edge":"18", "firefox":"68", "ie":"11" }
  proposal-unicode-property-regex { "chrome":"49", "edge":"18", "firefox":"68", "ie":"11" }
  transform-named-capturing-groups-regex { "chrome":"49", "edge":"18", "firefox":"68", "ie":"11" }
  transform-async-to-generator { "chrome":"49", "ie":"11" }
  transform-exponentiation-operator { "chrome":"49", "ie":"11" }
  transform-template-literals { "ie":"11", "ios":"12", "safari":"12.1" }
  transform-literals { "ie":"11" }
  transform-function-name { "chrome":"49", "edge":"18", "ie":"11" }
  transform-arrow-functions { "ie":"11" }
  transform-classes { "ie":"11" }
  transform-object-super { "ie":"11" }
  transform-shorthand-properties { "ie":"11" }
  transform-duplicate-keys { "ie":"11" }
  transform-computed-properties { "ie":"11" }
  transform-for-of { "chrome":"49", "ie":"11" }
  transform-sticky-regex { "ie":"11" }
  transform-unicode-escapes { "ie":"11" }
  transform-unicode-regex { "chrome":"49", "ie":"11" }
  transform-spread { "ie":"11" }
  transform-destructuring { "chrome":"49", "ie":"11" }
  transform-block-scoping { "ie":"11" }
  transform-typeof-symbol { "ie":"11" }
  transform-new-target { "ie":"11" }
  transform-regenerator { "chrome":"49", "ie":"11" }
  transform-modules-commonjs { "chrome":"49", "edge":"18", "firefox":"68", "ie":"11", "ios":"12", "opera":"68", "safari":"12.1", "samsung":"10.1" }
  proposal-dynamic-import { "chrome":"49", "edge":"18", "firefox":"68", "ie":"11", "ios":"12", "opera":"68", "safari":"12.1", "samsung":"10.1" }
```

These are the npm version:

### FIRST:
```
{
  npm: '6.13.4',
  ares: '1.15.0',
  brotli: '1.0.7',
  cldr: '35.1',
  http_parser: '2.8.0',
  icu: '64.2',
  llhttp: '1.1.4',
  modules: '72',
  napi: '5',
  nghttp2: '1.39.2',
  node: '12.14.0',
  openssl: '1.1.1d',
  tz: '2019c',
  unicode: '12.1',
  uv: '1.33.1',
  v8: '7.7.299.13-node.16',
  zlib: '1.2.11'
}
```

### SECOND:
```
{
 '@f2e/cibd-ui-event': '1.0.0',
  npm: '6.14.5',
  ares: '1.15.0',
  brotli: '1.0.7',
  cldr: '35.1',
  http_parser: '2.8.0',
  icu: '64.2',
  modules: '64',
  napi: '4',
  nghttp2: '1.34.0',
  node: '10.16.0',
  openssl: '1.1.1b',
  tz: '2019a',
  unicode: '12.1',
  uv: '1.28.0',
  v8: '6.8.275.32-node.52',
  zlib: '1.2.11' 
}
```

**Reply by JLHwung** [reply]
As you can see in the first environment, there is no targets printed in the debug output, while the second prints `{ chrome: 48, ... }`. That means in the first environment, the `preset-env` is not supplied with any options, per https://babeljs.io/docs/en/babel-preset-env#targets
> if no targets are specified, @babel/preset-env will transform all ECMAScript 2015+ code by default. We don't recommend using preset-env this way because it doesn't take advantage of its ability to target specific browsers.

**Reply by philkau** [reply]
Hi @JLHwung

Thanks for your reply~

I try the ### THIRD environment, the built result is different from the first.
Should I check the global setting or something?

**.babelrc**
```json
{
   "presets": [
   "@babel/preset-env",
       ["@babel/preset-env", { "debug": true }],
       "@babel/preset-react"
   ],
   "plugins": [
       "@babel/plugin-proposal-object-rest-spread"
   ]
}
```


**Result After Built:** (Different from the FIRST)

exports["default"] = void 0;


**Console:** (The same with the FIRST)

Using plugins:
  proposal-nullish-coalescing-operator {}
  proposal-optional-chaining {}
  proposal-json-strings {}
  proposal-optional-catch-binding {}
  transform-parameters {}
  proposal-async-generator-functions {}
  proposal-object-rest-spread {}
  transform-dotall-regex {}
  proposal-unicode-property-regex {}
  transform-named-capturing-groups-regex {}
  transform-async-to-generator {}
  transform-exponentiation-operator {}
  transform-template-literals {}
  transform-literals {}
  transform-function-name {}
  transform-arrow-functions {}
  transform-block-scoped-functions {}
  transform-classes {}
  transform-object-super {}
  transform-shorthand-properties {}
  transform-duplicate-keys {}
  transform-computed-properties {}
  transform-for-of {}
  transform-sticky-regex {}
  transform-unicode-escapes {}
  transform-unicode-regex {}
  transform-spread {}
  transform-destructuring {}
  transform-block-scoping {}
  transform-typeof-symbol {}
  transform-new-target {}
  transform-regenerator {}
  transform-member-expression-literals {}
  transform-property-literals {}
  transform-reserved-words {}
  transform-modules-commonjs {}
  proposal-dynamic-import {}

**Reply by philkau** [reply]
Hi @JLHwung

I remove the globel babel-cli in the FIRST environment.
The result is changed to **exports["default"] = void 0;**, the same as the SECOND and THIRD.
The new line signal become LF from CRLF.

And I install the global babel-cli back, the result is still **exports["default"] = void 0;**.

**Reply by JLHwung** [reply]
As I have said before, this is expected when _no_ target is provided.

Note that you can always supply a `defaults` targets to `preset-env`

```json
{
   "presets": [
   "@babel/preset-env",
       ["@babel/preset-env", { "targrets": "defaults" }],
       "@babel/preset-react"
   ]
}
```

You can also remove `"@babel/plugin-proposal-object-rest-spread"` and `preset-env` will determine whether it should be enabled from the targets config.
